### ISSUE TITLE
segfault possibly caused by prpltwtr

### ISSUE THREAD

**Issue by GoogleCodeExporter** [marig345/prpltwtr#11]
```
What steps will reproduce the problem?

unknown..  I woke up to a segfaulted finch..  finch has not segfaulted for
me in months, and I just started using prpltwtr about a week ago.. 
mbpurple used to segfault often, in earlier releases of that package..  I
notice the owner of prpltwtr is/was also a commiter for that package, so
possibly some shared code in both projects is causing the issue..

I'm not sure prpltwtr is even to blame, or how to debug, but here is some
info that may be useful:

xbj@Kilo:~$ dmesg | grep finch
finch[26692]: segfault at 696c6e73 ip b7451368 sp bfbc3150 error 4 in
libpurple.so.0.6.6[b7424000+eb000]



What version of the product are you using? On what operating system?

0.5.0, ArchLinux


```

Original issue reported on code.google.com by `xbj9...@gmail.com` on 17 Mar 2010 at 12:02

**Comment by GoogleCodeExporter** [comment]
```
Fair enough. I'll try running the plugin on multiple machines over a few nights 
under
debug mode to see what happens. I know there's probably an issue when receiving
properly formed xml with invalid data from twitter, but there may be another 
issue,
so we'll see what happens.
```

Original comment by `neav...@gmail.com` on 17 Mar 2010 at 1:26

**Comment by GoogleCodeExporter** [comment]
```
I believe I have found the problem and am testing the fix. Should be out 
sometime
this weekend hopefully. 

To be a bit technical, this is being caused when there is either a twitter 
disconnect
(either rate limit exceeded or a network issue) and there are multiple requests
pending (ie, get mentions and and home timelime). When the first request comes 
back,
it causes the disconnect (properly). The second request then tries to access the
connection causing a segfault. 
```

Original comment by `neav...@gmail.com` on 19 Mar 2010 at 7:47
- Changed state: **Started**
- Added labels: **Milestone-Release0.5.1**

**Comment by GoogleCodeExporter** [comment]
```
glad to be of help..  :)

I have also become the prpltwtr package maintainer for ArchLinux..
```

Original comment by `xbj9...@gmail.com` on 20 Mar 2010 at 1:36

**Comment by GoogleCodeExporter** [comment]
```
Has the issue reoccurred using the latest version?

Any other issues compiling on ArchLinux that I should be aware of? 
```

Original comment by `neav...@gmail.com` on 27 Mar 2010 at 8:03

**Comment by GoogleCodeExporter** [comment]
```
I haven't had another segfault since that first one..

and no, I've not had any problems with the Arch package.  :)
```

Original comment by `xbj9...@gmail.com` on 27 Mar 2010 at 8:35

**Comment by GoogleCodeExporter** [comment]
```
I had segfaults in pidgin as well when combining IRC and prpltwtr (Ubuntu 9.10,
upgraded to pidgin 2.6.6 and libpurple 2.6.6, prpltwtr 0.5.1)

- prpltwtr runs fine with most other accounts (xmpp, yahoo, skype) but not with 
IRC 
- eventually found out that it crashes as soon as the plugin "chat user list 
logging"
kicks in: part of the pidgin-plugin-pack on Ubuntu
(https://launchpad.net/ubuntu/karmic/sparc/pidgin-plugin-pack/2.6.0-0ubuntu1). 
it
lists (and logs) the users in a channel when joining.
- enabling the "chat user list logging" plugin only after pidgin started (with 
-n to
prevent logins) and then logging in to IRC does not create a segfault

(perhaps this should be an entirely different issue, but this seemed a logical 
place
to mention it for now)
```

Original comment by `rolfkl...@gmail.com` on 28 Mar 2010 at 8:55

**Comment by GoogleCodeExporter** [comment]
```
This should be a separate issue, but it looks like there's a conflict between
gtkprpltwtr plugin and the listlog plugin. Not sure what the issue is yet, 
since I
haven't taken a look at the debug logs yet, but should be a simple fix I 
believe.

For the time being, disabling either gtkprpltwtr or listlog should prevent the 
issue
while still allowing you to use both irc and prpltwtr.
```

Original comment by `neav...@gmail.com` on 28 Mar 2010 at 9:19

**Comment by GoogleCodeExporter** [comment]
```
Try this attached patch. 
Go to the prpltwtr source directory and run:

patch -p1 -i convicon_irc_fix.patch

Then recompile. Should fix the issue. 
I'll upload the full source tomorrow morning as v0.5.2
```

Original comment by `neav...@gmail.com` on 28 Mar 2010 at 10:07

Attachments:
- [convicon_irc_fix.patch](https://storage.googleapis.com/google-code-attachments/prpltwtr/issue-11/comment-8/convicon_irc_fix.patch)

**Comment by GoogleCodeExporter** [comment]
```
for what it's worth, I've always used prpltwtr in finch alongside IRC..
```

Original comment by `xbj9...@gmail.com` on 28 Mar 2010 at 10:14

**Comment by GoogleCodeExporter** [comment]
```
hello, and I tried the executable from XP and before passing through the 
Virustotal,
and are positive.

http://www.virustotal.com/analisis/4d158abf21db4c11a55d074819de08e6076713e50044c
e2f2c251fd7760d6100-1272260910
(MD5: 82dc2cbe9a8fa9ebc2f9e8e563062b46)

http://www.virustotal.com/analisis/f938ddb744bec5bdb2245e2b4ba887445fd4b64f92cf5
f22783f09ddb8df5079-1272141597
(MD5: 1d60124f919127675c1aee06c33d36ba)
```

Original comment by `diego.op...@gmail.com` on 27 Apr 2010 at 5:30

**Comment by GoogleCodeExporter** [comment]
Original comment by `mikeage` on 14 Feb 2011 at 6:37
- Changed state: **Verified**
